slapo/utils/report.py

```diff
@@ -14,13 +14,11 @@ def report_memory(msg="", report_gc=False):
         f"{msg} CPU RAM used: {psutil.virtual_memory()[3] / 1024 / 1024 / 1024:.4f} GiB"
     )
     if not dist.is_initialized():
-        return
+        return 0.0
```

Comment:
It seems a bit weird to see 0 memory usage in the single device environment? We could still print the memory usage without rank.